src/main/java/org/dataloader/DataLoaderHelper.java

```diff
@@ -134,7 +134,16 @@ Object getCacheKey(K key) {
                 loaderOptions.cacheKeyFunction().get().getKey(key) : key;
     }
 
-    CompletableFuture<List<V>> dispatch() {
+    public static class DispatchResult<X> {
+        public final CompletableFuture<List<X>> futureList;
```

Comment:
I know I know Java should have data classes so we dont have to write so much code

but it does - so I would (and will) use POJO patterns with getters.